status_bar_view.py

```diff
@@ -58,30 +68,45 @@ def on_selection_modified_async(self, _primary_view_):
             draw(**State)
 
 
-def draw(active_view, we_count, current_pos, **kwargs):
-    vid = active_view.id()
-
+def draw(active_view, we_count, current_pos, errors_per_line, **kwargs):
     if not we_count:
         active_view.erase_status(STATUS_KEY)
         return
 
-    status = "W: {warning} E: {error}".format(**we_count)
+    status = "W: {} E: {}".format(*we_count)
 
-    msgs = []
-    errors_on_pos = persist.errors.get_region_dict(vid, *current_pos)
-    for error_type, dc in errors_on_pos.items():
-        for d in dc:
-            msgs.append(d["msg"])
+    msgs = messages_under_cursor(errors_per_line, current_pos)
     if msgs:
         status += " - {}".format("; ".join(msgs))
 
     if status != active_view.get_status(STATUS_KEY):
         active_view.set_status(STATUS_KEY, status)
 
 
-def get_we_count(vid):
-    view_errors = persist.errors.get_view_dict(vid) if vid else {}
-    return view_errors.get('we_count_view', {})
+def messages_under_cursor(errors, current_pos):
+    line, col = current_pos
+    return [
+        error['msg'] for error in errors[line]
+        if error["start"] <= col <= error["end"]
+    ]
+
+
+def errors_per_line(errors):
+    rv = defaultdict(list)
```

Comment:
what's an rv?

![3689-jayco2017_precept31ul_34front_oceanblue](https://user-images.githubusercontent.com/2543659/35323485-ef726916-00ed-11e8-80f9-b0873e87ecdf.png)

🤔

Reply:
Hm, that's the common name for `return value`?

Reply:
Done

Reply:
> Hm, that's the common name for return value?

I guessed that but haven't seen it used like that much. It's all good 👍

Reply:
Interesting. I always use `ret` or `ret_val` for that.

Reply:
I think I have this from Armin Ronacher. ... yeaaars ago